cibuildwheel/oci_container.py

```diff
@@ -108,7 +108,16 @@ def __enter__(self) -> OCIContainer:
         if detect_ci_provider() == CIProvider.travis_ci and platform.machine() == "ppc64le":
             network_args = ["--network=host"]
 
-        shell_args = ["linux32", "/bin/bash"] if self.simulate_32_bit else ["/bin/bash"]
+        simulate_32_bit = self.simulate_32_bit
+        container_machine = call(
+            self.engine.name, "run", "--rm", self.image, "uname", "-m", capture_stdout=True
+        ).strip()
+        if container_machine not in {"x86_64", "aarch64"}:
+            # either the architecture running the image is already the right one
+            # or the image entrypoint took care of this
```

Comment:
I think that the ENTRYPOINT isn't used when you specify a program via `run` or `create`?